### PR TITLE
ALFREDAPI-489 Add contentIndexing check to SolrTestHelper

### PR DESCRIPTION
**Pull request by RVanhuysseXenit**
Fixes https://xenitsupport.jira.com/browse/ALFREDAPI-489

- [x] Is [CHANGELOG.md](https://github.com/xenit-eu/alfred-api/blob/master/CHANGELOG.md) extended?
- [N/A] Does this PR avoid breaking the API? 
    Breaking changes include adding, changing or removing endpoints and/or JSON objects used in requests and responses.
- [N/A] Does the PR comply to REST HTTP result codes policy outlined in the [user guide](https://docs.xenit.eu/alfred-api/stable-user/rest-api/index.html#rest-http-result-codes)?
- [N/A] Is error handling done through a method annotated with `@ExceptionHandler` in the webscript classes?
- [N/A] Does the PR follow our [coding styleguide and other active procedures](https://xenitsupport.jira.com/wiki/spaces/XEN/pages/624558081/XeniT+Enhancement+Proposals+XEP)?
- [N/A] Is usage of `this.` prefix avoided?

See [README.md](./README.md) for full explanation.

### CODE REVIEW

**Review by vierbergenlars — CHANGES_REQUESTED** [review]
That this worked is probably coincidence.

The property is given without `GLOBAL_` prefix.
Upon further investigation, there is no global property to configure arbitrary subsystems to autostart, and the Transformers bean does not have a configurable `autoStart` property (https://github.com/Alfresco/alfresco-community-repo/blob/5ce179f209d4be995940e9248f4961fb7eab639d/repository/src/main/resources/alfresco/bootstrap-context.xml#L289) .

File: apix-integrationtests/src/main/java/eu/xenit/apix/tests/search/TermHitHighlightingTest.java [hunk 1]
```diff
@@ -77,8 +85,9 @@ public void searchResponseContainsHighlights() throws IOException, InterruptedEx
                 + "with the !PREFIX!furies!SUFFIX!. A mirror’s temperature is always at zero.\n\n")));
 
         // Waiting for Solr's indexing process to catch up before executing test.
-        solrHelper.waitForSolrSync();
+        solrHelper.waitForTxnSync();
         SearchServiceTest.waitAWhile(20);
```
Comment (evyrosseels) [inline apix-integrationtests/src/main/java/eu/xenit/apix/tests/search/TermHitHighlightingTest.java:89]: Is this wait still needed now that there's another wait after it?

File: apix-integrationtests/52/src/main/java/eu/xenit/apix/util/SolrTestHelperImpl.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    @Override
+    public int getFtsStatusCleanDocs() {
+        JSONObject solrSummary = solrAdminClient.getSolrSummaryJson();
+        return (Integer) solrSummary.getJSONObject(SOLR_SUMMARY_CORE_ALFRESCO).getJSONObject(SOLR_SUMMARY_FTS).get(SOLR_SUMMARY_FTS_CLEAN);
+    }
+
+    @Override
+    public boolean isContentIndexed() {
+        JSONObject solrSummary = solrAdminClient.getSolrSummaryJson();
+        JSONObject ftsBlock = solrSummary.getJSONObject(SOLR_SUMMARY_CORE_ALFRESCO).getJSONObject(SOLR_SUMMARY_FTS);
```
Comment (evyrosseels) [inline apix-integrationtests/52/src/main/java/eu/xenit/apix/util/SolrTestHelperImpl.java:34]: Minor, but you could prevent some duplicate code by putting getting this fts block into its own method.

File: CHANGELOG.md [hunk 3]
```diff
@@ -10,6 +10,7 @@
 ### Deleted
 
 ### Fixed
+* [ALFREDAPI-489](https://xenitsupport.jira.com/browse/ALFREDAPI-489): start transformation subsystem at startup of integration tests.
```
Comment (evyrosseels) [inline CHANGELOG.md:13]: Might need to update this now that the solution is a little different

**Review by vierbergenlars — APPROVED** [review]

**Comment by vierbergenlars** [comment]
I agree with the changes suggested by Evy, but they are non-blocking in my opinion.

File: apix-integrationtests/52/src/main/java/eu/xenit/apix/util/SolrTestHelperImpl.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    private final String SOLR_SUMMARY_FTS_NEW = "Node count with FTSStatus New";
+    private final String SOLR_SUMMARY_FTS_DIRTY = "Node count with FTSStatus Dirty";
+    private final String SOLR_SUMMARY_FTS_CLEAN = "Node count with FTSStatus Clean";
+
+    public SolrTestHelperImpl(String baseUrl, DataSource dataSource, SwitchableApplicationContextFactory searchSubSystem) {
+        super(baseUrl, dataSource, searchSubSystem);
+    }
+
+    @Override
+    public int getFtsStatusCleanDocs() {
```
Comment (WimCrols) [inline apix-integrationtests/52/src/main/java/eu/xenit/apix/util/SolrTestHelperImpl.java:26]: This method should be moved to the superclass.

File: apix-integrationtests/src/main/java/eu/xenit/apix/util/SolrTestHelperBaseImpl.java [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        solrAdminClient = new SolrAdminClient(baseUrl, searchSubSystem);
+    }
+
+    /**
+     * Beware of calling this method too frequently since it runs an aggregate query over the entire alf_nodes table
+     *
+     * @return true if the latest Alfresco transaction has been indexed by Solr
+     */
+    @Override
+    public boolean areTxnsSynced() {
```
Comment (WimCrols) [inline apix-integrationtests/src/main/java/eu/xenit/apix/util/SolrTestHelperBaseImpl.java:35]: (non-blocking) I vastly prefer `Transaction` over `Tx` in all methods and variables.

File: apix-integrationtests/52/src/main/java/eu/xenit/apix/util/SolrTestHelperImpl.java (hunk 4 above)
Comment (WimCrols) [reply apix-integrationtests/52/src/main/java/eu/xenit/apix/util/SolrTestHelperImpl.java:26]: The other methods could be slimmed down a bit further too, but there is a tradeoff in legibility <-> maintenance.

**Review by WimCrols — CHANGES_REQUESTED** [review]

**Review by evyrosseels — APPROVED** [review]

File: apix-integrationtests/src/main/java/eu/xenit/apix/tests/search/TermHitHighlightingTest.java [hunk 6]
```diff
@@ -58,7 +63,10 @@ public void createHighlightTestTxt() throws InterruptedException {
             nodeService.setContent(target, inputStream, txtFile.getName());
             return null;
         }, false, true);
-        solrHelper.waitForSolrSync();
+        solrHelper.waitForTransactionSync();
+        solrHelper.waitForContentSync();
+        // Call bean representing the Transformers subsystem and start it. Subsystem is required for highlights.
+        transformersSubsystem.start();
```
Comment (vierbergenlars) [inline apix-integrationtests/src/main/java/eu/xenit/apix/tests/search/TermHitHighlightingTest.java:69]: I thought we would not start the transformers subsystem manually, but just wait for solr to have indexed the text content of documents?

Comment (RVanhuysseXenit) [reply apix-integrationtests/src/main/java/eu/xenit/apix/tests/search/TermHitHighlightingTest.java]: must have slipped through in previous pass. its gone now

**Review by vierbergenlars — DISMISSED** [review]

File: apix-docker/62/debug-extension.docker-compose.yml [hunk 7]
```diff
@@ -13,3 +13,4 @@ services:
     environment:
       - DEBUG=true
       - ALFRESCO_HOST=alfresco-core
+      - ALFRESCO_INTERNAL_HOST=alfresco-core
```
Comment (WimCrols) [inline apix-docker/62/debug-extension.docker-compose.yml:16]: Not needed for 7.0?

File: apix-integrationtests/52/src/main/java/eu/xenit/apix/util/SolrTestHelperImpl.java [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
```
Comment (WimCrols) [inline apix-integrationtests/52/src/main/java/eu/xenit/apix/util/SolrTestHelperImpl.java:24]: Wouldn't it be clearer to rename this to `getNumberOfFtsStatusCleanDocs`?

**Review by WimCrols — CHANGES_REQUESTED** [review]

**Review by WimCrols — APPROVED** [review]

**Pull request merged**